File: routes/flow/cartApi.js
var goods = require("./models/goods.js");
var res = require('./lib/resp.js');
var session = require('./lib/session.js');
var utils = require('./lib/utils.js');


if (request.getMethod() === 'GET') {
  var cart = session.getCart();
  var query = utils.getQuery();
  switch (query.a) {
    case 'add':
      var gs = goods.getByOid(query.oid);
      cart.add(gs.oid, gs.goods_name, gs.goods_img, gs.shop_price, 1);
      break;
    case 'del':
      cart.del(query.oid);
      break;
    case 'incr':
      cart.incr(query.oid);
      break;
    case 'decr':
      cart.decr(query.oid);
      break;
    default:
  }
  session.setCart(cart);
  res.redirect('./cart.js');
}

File: routes/flow/cartApi.go
package flow

import (
	"github.com/zengming00/go-testShop/framework"
	"github.com/zengming00/go-testShop/lib"
)

func CartApi(ctx *framework.HandlerContext) {
	if ctx.R.Method == "GET" {
		var cart = ctx.GetCart()
		var a = ctx.R.FormValue("a")
		var oid = ctx.R.FormValue("oid")
		switch a {
		case "add":
			var gs = ctx.Goods.GetByOid(oid)
			cart.Add(gs.Oid, gs.Goods_name, gs.Goods_img, lib.ParsePositiveInt(gs.Shop_price, 0), 1)
			break
		case "del":
			cart.Del(oid)
			break
		case "incr":
			cart.Incr(oid)
			break
		case "decr":
			cart.Decr(oid)
			break
		default:
		}
		ctx.SetCart(cart)
		ctx.Redirect("./cart.go")
	}
}
